examples/systemcalls/systemcalls.c

```diff
@@ -16,7 +34,22 @@ bool do_system(const char *cmd)
  *   and return a boolean true if the system() call completed with success
  *   or false() if it returned a failure
 */
-
+    int ret = system(cmd);
+    // check if system command was executed properly or not
+    if(ret == -1)
+    {
+    	syslog(LOG_ERR, "Child process could not be created or status cannot be retrieved. Return value = -1, errno = %d", errno);
+    	return false;
+    }
+    
+    // return 127 if a shell could not be executed in the child process
+    if(ret == 127)
+    {
+    	syslog(LOG_ERR, "Shell could not be executed in the child process then return value is as though the child shell terminated by calling exit(1), errno = %d", errno);
+    	return false;
+    }
+    
+    syslog(LOG_DEBUG, "system command successfully executed");	
```

Comment:
@abaa3023 I think you are missing a check for nonzero return value here, should use `WEXITSTATUS()` to get this and verify it is 0 before returning true.